### PR TITLE
Revert: refactor `Autocomplete` component to pass `exhaustive-deps`

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Reverts the recent changes in #41382

## Why?
This change introduced two new bugs and we're still working on fixes for them:
- #41724 
- #41709 

Reverting the change, adding some tests, and retrying the refactor with these obstacles in mind will be a better UX than keeping the bugs while working on a fix.

### CODE REVIEW

**Review by ciampo — APPROVED** [review]
Code changes LGTM

I just wonder if we should add ESLint ignore comments pointing to this PR to explain why the exhaustive-deps rule is not currently enforced for this component.

I did a quick test and the issues described in #41709 and #41724 can no longer be reproduced — @chad1008 can you also confirm? If that's the case, we can close these 2 issues when this PR gets merged.

**Comment by chad1008** [comment]
> I just wonder if we should add ESLint ignore comments pointing to this PR to explain why the exhaustive-deps rule is not currently enforced for this component.

Good idea. I don't plan on leaving them this way long (I really want to get a better pass at this one, asap!) but smart to leave some context in the meantime!

> I did a quick test and the issues described in https://github.com/WordPress/gutenberg/issues/41709 and https://github.com/WordPress/gutenberg/issues/41724 can no longer be reproduced — @chad1008 can you also confirm? If that's the case, we can close these 2 issues when this PR gets merged.

Confirmed. I tried as hard as I could and couldn't get either to reproduce. I'll close both issues out when this revert merges 👍

File: packages/editor/src/hooks/default-autocompleters.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { userAutocompleter } from '../components';
 function setDefaultCompleters( completers = [] ) {
 	// Provide copies so filters may directly modify them.
 	completers.push( clone( userAutocompleter ) );
-
+	completers.push( clone( { ...userAutocompleter, triggerPrefix: '+' } ) );
```
Comment (ciampo) [inline packages/editor/src/hooks/default-autocompleters.js]: I believe this line should not be part of the PR, probably a leftover from testing for any regressions

Comment (chad1008) [reply packages/editor/src/hooks/default-autocompleters.js]: 😱 That's embarrassing, thank you very very much.

Comment (ciampo) [reply packages/editor/src/hooks/default-autocompleters.js]: Don't worry, it's totally fine! That's why we code-review our changes :)

**Comment by ciampo** [comment]
e2e tests keep failing, maybe a rebase will solve the issue

**Comment by chad1008** [comment]
Rebased and waiting on tests again 🤞
